NuRadioReco/modules/io/eventReader.py

```diff
@@ -9,8 +9,22 @@ class eventReader:
     read events from file
     """
 
-    def begin(self, filename, log_level=logging.WARNING):
-        self.__fin = NuRadioRecoio.NuRadioRecoio(filename, parse_header=False, log_level=log_level)
+    def begin(self, filename, read_detector=False, log_level=logging.WARNING):
+        """
+        Setup function for the eventReader module
+
+        Parameters
+        -----------
+        filename: array if strings
+            names of the input files
+        read_detector: boolean
+            If True, the eventReader will parse the detector description in the
+            event files and make them accessible through the get_detector function,
+            assuming the file contains a detector
+        log_level: logging enum
+        """
+
+        self.__fin = NuRadioRecoio.NuRadioRecoio(filename, parse_header=read_detector, log_level=log_level)
```

Comment:
Isn't this a littel confusing? If parse_header is like read_detector?

Reply:
Or I just don't understand the sentence in the documentation.

Reply:
Calling it parse_header did seem more confusing to me, since the headers are not accessible from the eventReader anyway. Reading the detector from the file is done in the same function that parses the header (I didn't see any good reason to split this and loop twice), so thats where this naming is coming from.

Reply:
but setting this to true actually also parses the header right? And the event reader can also return the header. So it actually does both. This should be reflected at least in the doc string.

Reply:
I updated the docstring to make this clearer